addons/website_blog/models/website_blog.py

```diff
@@ -254,6 +257,9 @@ def write(self, cr, uid, ids, vals, context=None):
             ids = [ids]
         if 'content' in vals:
             vals['content'] = self._postproces_content(cr, uid, ids[0], vals['content'], context=context)
+        post = self.browse(cr, uid, ids, context)
+        if 'website_published' in vals and vals['website_published'] and post.create_date == post.date_published:
+            vals['date_published'] = datetime.now()
```

Comment:
@rga-odoo ,
If the post has not been published yet, the published date by default is creation date. When a post is unpublished and then published again, the publish date should be updated to the last time it is published